langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchEngine.java

```diff
@@ -0,0 +1,45 @@
+package dev.langchain4j.web.search;
+
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+/**
+ * Represents a web search engine that can be used to perform searches on the Web in response to a user query.
+ */
+public interface WebSearchEngine {
+
+    /**
+     * Runs a search query on the web search engine and returns a pretty-string representation of the search results.
+     *
+     * @param query the search query
+     * @return a pretty-string representation of the search results
+     */
+    default String runSearch(String query) {
+        return search(query).results()
+                .stream()
+                .map(organicResult -> organicResult.title() + "\n" + organicResult.content() + "\n" + "Source: " + organicResult.link())
+                .collect(Collectors.joining("\n\n"));
+    }
+
+    /**
+     * Performs a search query on the web search engine and returns the search results.
+     *
+     * @param query the search query
+     * @return the search results
+     * @throws IllegalArgumentException if no web search results are found for the query
+     */
+    default WebSearchResults search(String query) {
+        WebSearchResults results = search(WebSearchRequest.from(query));
+        ensureNotNull(results, "No web search results found for query: " + query);
```

Comment:
Is it possible to get no results at all?

Reply:
No, maybe the value of the name is confusing but I wanted to validate that if there is no result something went wrong and an exception could be thrown to the user

Reply:
minor: I would move this check to `WebSearchRetults` constructor, so that it even canot be created empty